test/functional/interface_ipc.py

```diff
@@ -204,13 +208,32 @@ async def async_routine():
             res = await template.result.submitSolution(ctx, block.nVersion, block.nTime, block.nNonce, coinbase.serialize())
             assert_equal(res.result, False)
             self.log.debug("Submit a valid block")
+            remote_block_before = await self.parse_and_deserialize_block(template, ctx)
             block.nVersion = original_version
             block.solve()
+
+            self.log.debug("First call checkBlock()")
             res = await mining.result.checkBlock(block.serialize(), check_opts)
             assert_equal(res.result, True)
+
+            self.log.debug("Submitted coinbase must include witness")
+            assert_not_equal(coinbase.serialize_without_witness().hex(), coinbase.serialize().hex())
+            res = await template.result.submitSolution(ctx, block.nVersion, block.nTime, block.nNonce, coinbase.serialize_without_witness())
```

Comment:
In commit 504acfcd5d1ed4d77f7fd4f5dc57e66d6718d3d4:

If I apply this test independently before the miner.cpp code changes, it still passes, so not sure what we are testing here. Wouldn't this only be relevant if we'd run checkBlock on the template's own block? Afaict we only do checkBlock on blocks we re-create here.

Reply:
Strange, for me this test fails as expected with the following diff applied to this branch:

```diff
diff --git a/src/node/miner.cpp b/src/node/miner.cpp
index b988e28a3f..485984e437 100644
--- a/src/node/miner.cpp
+++ b/src/node/miner.cpp
@@ -454,9 +454,9 @@ void AddMerkleRootAndCoinbase(CBlock& block, CTransactionRef coinbase, uint32_t
     block.hashMerkleRoot = BlockMerkleRoot(block);

     // Reset cached checks
-    block.m_checked_witness_commitment = false;
-    block.m_checked_merkle_root = false;
-    block.fChecked = false;
+    // block.m_checked_witness_commitment = false;
+    // block.m_checked_merkle_root = false;
+    // block.fChecked = false;
 }
```